### ISSUE TITLE
[Astro + Svelte + Cloudflare] Cannot build Svelte component that contains $onMount() with Cloudflare adapter

### ISSUE THREAD

**Issue by GoodLucky777** [withastro/adapters#485]
### Astro Info

```block
Astro                    v5.0.5
Node                     v18.20.3
System                   Linux (x64)
Package Manager          unknown
Output                   static
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/svelte
```

### Describe the Bug

If I try to build the Astro website that contains Svelte component with $onMount() and Cloudflare adapter, the build error will occurs like this.

Build error:

```
$ npm run build

prerendering static routes 
01:36:03 ▶ src/pages/index.astro
01:36:03   └─ /index.htmlhttps://svelte.dev/e/lifecycle_outside_component
  Stack trace:
    at lifecycle_outside_component (file:///home/projects/github-memgm1vc/dist/_worker.js/pages/index.astro.mjs?time=1734280563043:32:9)
    at Test (file:///home/projects/github-memgm1vc/dist/_worker.js/pages/index.astro.mjs?time=1734280563043:69:9)
    at renderToStaticMarkup (file:///home/projects/github-memgm1vc/dist/_worker.js/chunks/_@astro-renderers_BFYr5CxZ.mjs:1251:23)
```

It doesn't occurs when I use NodeJS adapter. It seems it is occurring after `@astrojs/cloudflare@12.1.0` update.

### What's the expected result?

No build errors.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/astro-svelte-cloudflare-build-error

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
Can you try Svelte 5.15.0 in your project if it fixes the issue? I've sent a change to add the worker condition, so it should work better now. If it doesn't cause other build tooling regressions that requires a revert there, I think it would be the best fix given the case here.

**Comment by GoodLucky777** [comment]
> Can you try Svelte 5.15.0 in your project if it fixes the issue? I've sent a change to add the worker condition, so it should work better now. If it doesn't cause other build tooling regressions that requires a revert there, I think it would be the best fix given the case here.

It seems the build error is fixed in Svelte 5.15.0.

**Comment by bluwy** [comment]
I'll close this for now as fixed then. Thanks for testing!
